Implementing Auto Align

ChaoticChaosTheGreat opened the pull request:
Implementing Auto Align

ChaoticChaosTheGreat commented:
Design Doc: https://www.notion.so/warriorborgs/Auto-Align-b19ec3943dfa43a5b86a708d382986c5

ChaoticChaosTheGreat commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -20,6 +21,17 @@
 
 public final class Constants {
 
+  public static final boolean kAlliance = false; // true = blue, false = red
```
Need to add some sort of interface to change this value in shuffleboard.

3LucasZ replied:
you can change the value during sim. Look at the alliance color, and check blue/red in the GUI. This constant is a terrible way of assigning this value. II can help you with this on Friday if you remind me.

3LucasZ commented:
Get this working in sim and send an image/video, please.

3LucasZ commented on src/main/deploy/pathplanner/paths/New Path.path:
```diff
@@ -0,0 +1,65 @@
+{
+  "version": 1.0,
+  "waypoints": [
+    {
```
why make a new path?

ChaoticChaosTheGreat replied:
Sorry, it was used to get the waypoints for dpg, will delete.

3LucasZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -39,6 +41,8 @@ public class RobotContainer {
   private final CommandXboxController driver = new CommandXboxController(0);
   private final CommandXboxController operator = new CommandXboxController(1);
 
+  private final Joystick joystick = new Joystick(1);
```
Please, do not use Joystick. Please, use CommandXboxController (drive or operator, or make new one called tester if you really need). Please, do not use Joystick.

ChaoticChaosTheGreat replied:
Ye again sorry, I didn't know how to get the auto-align command to run in sim so I was trying some new things.

3LucasZ commented on src/main/java/frc/robot/autos/PathPlannerAuto.java:
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2024 FRC 3256
+// https://github.com/Team3256
```
Make a comment explaining what this file does. Also, looks very repetitive (something we have implemented before). Is this just Auto Chooser and Runner?

ChaoticChaosTheGreat replied:
This file needs to be deleted.

3LucasZ commented on src/main/java/frc/robot/autos/pathgeneration/PathGeneration.java:
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2024 FRC 3256
+// https://github.com/Team3256
+//
+// Use of this source code is governed by a 
```
empty class? remove if this is just a placeholder?

3LucasZ commented on src/main/java/frc/robot/autos/pathgeneration/commands/AutoAmp.java:
```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2024 FRC 3256
```
rename file to something better please

3LucasZ commented on src/main/java/frc/robot/autos/pathgeneration/commands/AutoAmp.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    PathConstraints constraints =
+        new PathConstraints(3.0, 4.0, Units.degreesToRadians(540), Units.degreesToRadians(720));
```
make these constants. the first rule is to not use numbers out in the open like this. make descriptive constants, please.

3LucasZ commented on src/main/java/frc/robot/autos/pathgeneration/commands/AutoAmp.java:
```diff
@@ -0,0 +1,55 @@
+        new PathConstraints(3.0, 4.0, Units.degreesToRadians(540), Units.degreesToRadians(720));
+
+    Command moveToScoringLocation =
+        AutoBuilder.pathfindToPose(scoringLocation, constraints, 0.0, 0.0);
```
tell us what 0.0 and 0.0 means through a constant in the constant file, please

3LucasZ commented on src/main/java/frc/robot/autos/pathgeneration/commands/AutoAmp.java:
```diff
@@ -0,0 +1,55 @@
+    Command autoScore = Commands.sequence(moveToScoringLocation);
+
+    addChildCommands(autoScore);
+    super.initialize();
```
why calling initialize in the bottom?

ChaoticChaosTheGreat replied:
Running the parentcommand initialize to add the child commands.

3LucasZ commented on src/main/java/frc/robot/autos/pathgeneration/commands/AutoSpeaker.java:
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2024 FRC 3256
+// https://github.com/Team3256
```
need better filename

3LucasZ commented on src/main/java/frc/robot/autos/pathgeneration/commands/AutoSpeaker.java:
```diff
@@ -0,0 +1,57 @@
+
+    //        Pose2d start = swerveSubsystem.getPose();
+    //
+    if (kAlliance == false) {
```
please, use Robot.getAlliance feature instead of this. I can help you with this on friday if you ask me

3LucasZ commented on src/main/java/frc/robot/autos/pathgeneration/commands/AutoSpeaker.java:
```diff
@@ -0,0 +1,57 @@
+    System.out.println("Scoring Location: " + scoringLocation);
+
+    PathConstraints constraints =
+        new PathConstraints(3.0, 4.0, Units.degreesToRadians(540), Units.degreesToRadians(720));
```
please, do not use numbers out in the open like this and use constants

3LucasZ commented on src/main/java/frc/robot/autos/pathgeneration/helpers/ParentCommand.java:
```diff
@@ -0,0 +1,46 @@
+    for (Command childCommand : childCommands) {
+      System.out.println("Parent killing child " + childCommand.getName());
+      childCommand.cancel();
+    }
```
May not be necessary, will take a look again after you make requested changes.

ChaoticChaosTheGreat commented:
Removed ParentCommand (unnecessary ), made all numbers constants with AutoConstants class, fixed bug in sim which didn't let you run the command twice, removed alliance constant and replaced with retrieving the alliance from he DS.

3LucasZ commented:
great job.

3LucasZ reviewed (approved):
Very good job, @ChaoticChaosTheGreat (Ryan).

3LucasZ commented:
You have the honors to merge when you are ready.

ChaoticChaosTheGreat commented:
> Very good job, @ChaoticChaosTheGreat (Ryan).

Thanks!!! 😊
